src/contract/table-cursor.ts

```diff
@@ -63,113 +55,65 @@ export class TableCursor<TableRow> {
             return []
         }
 
-        let lower_bound = this.tableParams.lower_bound
-        const upper_bound = this.tableParams.upper_bound
+        let lower_bound = this.params.lower_bound
+        const upper_bound = this.params.upper_bound
 
         if (this.next_key) {
             lower_bound = this.next_key
         }
 
-        let indexPosition = this.tableParams.index_position || 'primary'
-
-        if (this.indexPositionField) {
-            const fieldToIndexMapping = this.table.getFieldToIndex()
-
-            if (!fieldToIndexMapping[this.indexPositionField]) {
-                throw new Error(`Field ${this.indexPositionField} is not a valid index.`)
-            }
-
-            indexPosition = fieldToIndexMapping[this.indexPositionField].index_position
-        }
-
-        const result = await this.table.contract.client!.v1.chain.get_table_rows({
-            ...this.tableParams,
-            limit: Math.min(
-                this.maxRows - this.rowsCount,
-                rowsPerAPIRequest || this.tableParams.limit
-            ),
+        const query = {
+            ...this.params,
+            limit: Math.min(this.maxRows - this.rowsCount, rowsPerAPIRequest || this.params.limit),
             lower_bound: wrapIndexValue(lower_bound),
             upper_bound: wrapIndexValue(upper_bound),
-            index_position: indexPosition,
-            json: false,
-        })
+        }
 
-        let {rows} = result
-        this.next_key = result.next_key
+        const result = await this.client!.v1.chain.get_table_rows(query)
 
+        const rows =
+            this.params.json === false && !(query as API.v1.GetTableRowsParamsTyped).type
+                ? result.rows.map((row) =>
+                      Serializer.decode({
+                          data: row,
+                          abi: this.abi,
+                          type: this.type,
+                      })
+                  )
+                : result.rows
+
+        this.next_key = result.next_key
         this.rowsCount += rows.length
 
         if (!result.next_key || rows.length === 0 || this.rowsCount === this.maxRows) {
             this.endReached = true
         }
 
-        rows = rows.map((row) =>
-            Serializer.decode({
-                data: row,
-                abi: this.table.contract.abi,
-                type: this.table.abi.type,
-            })
-        )
-
         return rows
     }
 
-    /**
-     * Resets the cursor to the beginning of the table and returns the first rows.
-     *
-     * @returns The first rows in the table.
-     */
     async reset() {
         this.next_key = undefined
         this.endReached = false
         this.rowsCount = 0
     }
 
-    /**
-     * Returns all rows in the cursor query.
-     *
-     * @returns All rows in the cursor query.
-     */
-    async all() {
-        const rows: TableRow[] = []
+    async all(): Promise<RowType[]> {
+        const rows: any[] = []
```

Comment:
This should be `const rows: RowType[] = []`, no?

Reply:
```suggestion
        const rows: RowType[] = []
```

Reply:
You're absolutely right.

Also, whatever this "Commit Suggestion" thing is, is awesome 😍 

It just let me click a button and merge that change in!

Reply:
Hehe yeah, it's pretty cool 😄